src/System.Drawing.Common/tests/Printing/PreviewPrintControllerTests.cs

```diff
@@ -0,0 +1,175 @@
+
+                var printEventArgs = new PrintPageEventArgs(null, Rectangle.Empty, Rectangle.Empty, new PageSettings());
+                Assert.NotNull(controller.OnStartPage(document, printEventArgs));
+
+                // Call OnEndPage.
+                controller.OnEndPage(document, printEventArgs);
+
+                // Call EndPrint.
+                controller.OnEndPrint(document, new PrintEventArgs());
+            }
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Fixed a NullReferenceException")]
+        public void OnStartPage_InvokeNullDocument_ThrowsArgumentNullException()
+        {
+            var controller = new PreviewPrintController();
+            var e = new PrintPageEventArgs(null, Rectangle.Empty, Rectangle.Empty, null);
+            Assert.Throws<ArgumentNullException>("document", () => controller.OnStartPage(null, e));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Fixed a NullReferenceException")]
+        public void OnStartPage_InvokeNullEventArgs_ThrowsArgumentNullException()
+        {
+            using (var document = new PrintDocument())
+            {
+                var controller = new PreviewPrintController();
+                Assert.Throws<ArgumentNullException>("e", () => controller.OnStartPage(document, null));
+            }
+        }
+
+        [ConditionalFact(Helpers.AnyInstalledPrinters, Helpers.IsDrawingSupported)]
+        public void OnStartPage_InvokeNullEventArgsPageSettings_ReturnsNull()
+        {
+            using (var document = new PrintDocument())
+            {
+                var controller = new PreviewPrintController();
+                controller.OnStartPrint(document, new PrintEventArgs());
+
+                var printEventArgs = new PrintPageEventArgs(null, Rectangle.Empty, Rectangle.Empty, null);
+                Assert.Throws<NullReferenceException>(() => controller.OnStartPage(document, printEventArgs));
+            }
+        }
+
+        [Fact]
+        public void OnStartPage_PrintNotStarted_ThrowsNullReferenceException()
+        {
+            using (var document = new PrintDocument())
+            {
+                var controller = new PreviewPrintController();
+                var e = new PrintPageEventArgs(null, Rectangle.Empty, Rectangle.Empty, null);
+                Assert.Throws<NullReferenceException>(() => controller.OnStartPage(document, e));
+            }
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void OnEndPage_InvokeWithoutStartingWindows_ThrowsNullReferenceException()
+        {
+            using (var document = new PrintDocument())
+            {
+                var controller = new PreviewPrintController();
+                Assert.Throws<NullReferenceException>(() => controller.OnEndPage(document, new PrintPageEventArgs(null, Rectangle.Empty, Rectangle.Empty, null)));
```

Comment:
I think we should be consistent and throw NullReferenceException on both, Unix and Windows, any reason why you didn't do it?